source/volume_of_fluid/assembler.cc

```diff
@@ -0,0 +1,667 @@
+/*
+ Copyright (C) 2016 - 2018 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/simulator_access.h>
+#include <aspect/utilities.h>
+#include <aspect/free_surface.h>
+#include <aspect/volume_of_fluid/utilities.h>
+#include <aspect/volume_of_fluid/field.h>
+#include <aspect/volume_of_fluid/assembly.h>
+
+//#include <deal.II/base/quadrature_lib.h>
+//#include <deal.II/lac/full_matrix.h>
+//#include <deal.II/lac/constraint_matrix.h>
+//#include <deal.II/grid/tria_iterator.h>
+//#include <deal.II/grid/filtered_iterator.h>
+//#include <deal.II/dofs/dof_accessor.h>
+//#include <deal.II/dofs/dof_tools.h>
+//#include <deal.II/fe/fe_values.h>
+
+namespace aspect
+{
+  namespace Assemblers
+  {
+    template <int dim>
+    void VolumeOfFluidAssembler<dim>::set_volume_fraction_threshold(const double value)
```

Comment:
Where is this ever called? If it is necessary, would it be possible to get this from the vof_handler, instead of storing it in this assembler (i.e. `this->get_volume_of_fluid_handler().get_volume_fraction_threshold()`)?